MSCloudLoginAssistant.psm1

```diff
@@ -36,32 +42,39 @@ function Test-MSCloudLogin
             $connectCmdletMfaRetryArgs = "";
             $variablePrefix = "az"
         }
-        "AzureAD"
+        'AzureAD'
         {
             $testCmdlet = "Get-AzureADUser";
             $exceptionStringMFA = "AADSTS";
             $connectCmdlet = "Connect-AzureAD";
-            $connectCmdletArgs = "-Credential `$o365Credential";
-            $connectCmdletMfaRetryArgs = "-AccountId `$o365Credential.UserName";
+            $connectCmdletArgs = "-Credential `$O365Credential";
+            $connectCmdletMfaRetryArgs = ""
             $variablePrefix = "aad"
         }
-        "SharePointOnline"
+        'SharePointOnline'
         {
-            $global:spoAdminUrl = Get-SPOAdminUrl;
+            if ($null -eq $ConnectionUrl)
+            {
+                $global:spoAdminUrl = Get-SPOAdminUrl;
+            }
+            else
+            {
+                $global:spoAdminUrl = $ConnectionUrl
+            }
             $testCmdlet = "Get-SPOSite";
             $exceptionStringMFA = "sign-in name or password does not match one in the Microsoft account system";
             $connectCmdlet = "Connect-SPOService";
-            $connectCmdletArgs = "-Url $global:spoAdminUrl -Credential `$o365Credential";
-            $connectCmdletMfaRetryArgs = $connectCmdletArgs.Replace("-Credential `$o365Credential","");
+            $connectCmdletArgs = "-Url $global:spoAdminUrl -Credential `$O365Credential";
             $variablePrefix = "spo"
```

Comment:
$connectCmdletMfaRetryArgs is used when we catch a login error related to MFA, why did we remove this?

Reply:
Again, they are empty, adding $connectCmdletMfaRetryArgs = "" back